docs/content/concepts/sui-move-concepts/conventions.mdx

```diff
@@ -0,0 +1,487 @@
+- `to_object_name`: Transforms an Object X to Object Y.
+- `from_object_name`: Transforms an Object Y to Object X.
+- `property_name`: Returns an immutable reference or a copy.
+- `property_name_mut`: Returns a mutable reference.
+
+## Potato Structs
+
+Do not call structs as Potato. It is a pattern recognized by the lack of abilities.
+
+```move
+module conventions::request {
+  // ✅ Right
+  struct Request {}
+
+  // ❌ Wrong
+  struct RequestPotato {}
+}
+```
+
+## Read Functions
+
+Be mindful of the dot syntax when naming functions. Avoid using the object name on function names.
+
+```move
+module conventions::lib {
+
+  struct Profile {
+    age: u64
+  }
+
+  // ✅ Right
+  public fun age(self: &Profile):  u64 {
+    self.age
+  }
+
+  // ❌ Wrong
+  public fun profile_age(self: &Profile): u64 {
+    self.age
+  }
+}
+
+module conventions::defi {
+
+  use conventions::lib::{Self, Profile};
+
+  public fun get_tokens(profile: &Profile) {
+
+  // ✅ Right
+  let name = profile.age();
+
+  // ❌ Wrong
+  let name2 = profile.profile_age();
+}
+```
+## Empty Function
+
+Functions that create data structures must be called empty.
+
+```move
+module conventions::collection {
+
+  struct Collection has copy, drop, store {
+    bits: vector<u8>
+  }
+
+  public fun empty(): Collection {
+    Collection {
+      bits: vector[]
+    }
+  }
+}
+```
+## New Function
+
+Functions that create objects must be called new.
+
+```move
+module conventions::object {
+
+  use sui::object::{Self, UID};
+  use sui::tx_context::TxContext;
+
+  struct Object has key, store {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Object {
+    Object {
+      id: object::new(ctx)
+    }
+  }
+}
+```
+
+## Shared Objects
+
+Library modules that share objects should provide two functions: one to create the object and another to share it. It allows the caller to access its UID and run custom functionality before sharing it.
+
+```move
+module conventions::profile {
+
+  use sui::object::{Self, UID};
+  use sui::tx_context::TxContext;
+  use sui::transfer::share_object;
+
+  struct Profile has key {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Profile {
+    Profile {
+      id: object::new(ctx)
+    }
+  }
+
+  public fun share(profile: Profile) {
+    share_object(profile);
+  }
+}
+```
+
+## Reference Functions
+
+Functions that return a reference must be named property_name_mut or property_name.
+
+```move
+module conventions::profile {
+
+  use std::string::String;
+
+  use sui::object::UID;
+
+  struct Profile has key {
+    id: UID,
+    name: String,
+    age: u8
+  }
+
+  // profile.name()
+  public fun name(self: &Profile): &String {
+    &self.name
+  }
+
+  // profile.age_mut()
+  public fun age_mut(self: &mut Profile): &mut u8 {
+    &mut self.age
+  }
+}
+```
+
+## Separation of Concerns
+
+Modules must be designed around one Object or Data Structure. A variant structure should have its own module to avoid complexity and bugs.
```

Comment:
```suggestion
Design your modules around one object or data structure. A variant structure should have its own module to avoid complexity and bugs.
```